server/src-lib/Data/Parser/JSONPath.hs

```diff
@@ -64,6 +61,36 @@ parseJSONPath = parseResult . parse parseElements
           Left $ invalidMessage remain
         else
           Right r
-
     invalidMessage s = "invalid property name: "  ++ T.unpack s
       ++ ". Accept letters, digits, underscore (_) or hyphen (-) only"
+      ++ ". Use single quotes enclosed in bracket if there are any special characters"
+
+-- available in Data.Aeson.Types but not importing
+formatPath :: JSONPath -> String
+formatPath path = "$" ++ formatRelativePath path
+
+formatRelativePath :: JSONPath -> String
+formatRelativePath path = format "" path
+  where
+    format :: String -> JSONPath -> String
+    format pfx []                = pfx
+    format pfx (Index idx:parts) = format (pfx ++ "[" ++ show idx ++ "]") parts
+    format pfx (Key key:parts)   = format (pfx ++ formatKey key) parts
+
+    formatKey :: T.Text -> String
+    formatKey key
+       | isIdentifierKey strKey = "." ++ strKey
+       | otherwise              = "['" ++ escapeKey strKey ++ "']"
+      where strKey = T.unpack key
+
+    isIdentifierKey :: String -> Bool
+    isIdentifierKey []     = False
+    isIdentifierKey (x:xs) = isAlpha x && all isAlphaNum xs
+
+    escapeKey :: String -> String
+    escapeKey = concatMap escapeChar
+
+    escapeChar :: Char -> String
+    escapeChar '\'' = "\\'"
+    escapeChar '\\' = "\\\\"
+    escapeChar c    = [c]
```

Comment:
We have `encodeJSONPath` function in `Hasura.RQL.Types.Error` module. How it is different from `formatPath`? If the later is more robust than the former, then please reuse `formatPath` in place of `encodeJSONPath`.

Reply:
I didn't know about the `encodeJSONPath` function, I think the `formatPath` function seems more robust so I'll use that